hooks/pre-exit

```diff
@@ -2,12 +2,16 @@
 
 set -eo pipefail
 
-WORKSPACE="$HOME/buildkite-checkouts/$BUILDKITE_BUILD_ID/$BUILDKITE_JOB_ID"
-CLONE_URL="${BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_CLONE_URL:-$BUILDKITE_REPO}"
-CLONE_DIR="${WORKSPACE}/$(echo "${CLONE_URL%/}" | awk -F '/' '{ print $NF }')"
-if [ -d "$CLONE_DIR" ]; then
-    echo "--- :broom: Cleaning up workspace"
-    echo "Removing directory: $CLONE_DIR"
-    rm -r "$CLONE_DIR"
+echo "--- :broom: Cleaning up workspace"
+echo "Removing directory: $WORKSPACE/$CLONE_DIR"
+if [ -d "$WORKSPACE/$CLONE_DIR" ]; then
+  sudo rm -r "$WORKSPACE/$CLONE_DIR"
+  unset CLONE_DIR
+  unset CLONE_URL
+fi
+
+echo "Removing workspace directory: $WORKSPACE"
+if [ -d "$WORKSPACE" ] && [ -z "$(ls -A "$WORKSPACE")" ]; then
+  rmdir "$WORKSPACE"
```

Comment:
Could we just do a `sudo rm -rf $WORKSPACE` instead of having a if condition that checks for non-empty repository? Because `rm -rf` automatically takes care of that logic and always yields 0 as exit code.

Reply:
changed as suggested